invenio_vocabularies/contrib/subjects/hierarchal_subject_vocabularies/gemet/datastreams.py

```diff
@@ -0,0 +1,125 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+#
+# Invenio-Vocabularies is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""GEMET subjects datastreams, readers, transformers, and writers."""
+
+import gzip
+import io
+
+from rdflib.namespace import RDFS
+
+from ...config import gemet_file_url
+from ..datastreams import (
+    BaseHierarchalSubjectsHTTPReader,
+    BaseHierarchalSubjectsTransformer,
+)
+
+
+class GEMETSubjectsHTTPReader(BaseHierarchalSubjectsHTTPReader):
+    """Reader class to fetch and process GEMET RDF data."""
+
+    def __init__(self, origin=None, mode="r", since=None, *args, **kwargs):
+        """Initialize the reader with the data source.
+
+        :param origin: The URL from which to fetch the RDF data.
+        :param mode: Mode of operation (default is 'r' for reading).
+        """
+        self.origin = origin or gemet_file_url
+        super().__init__(origin=self.origin, mode=mode, *args, **kwargs)
+
+    def _get_rdf_content(self, response):
+        """Decompress and return RDF content."""
+        with gzip.GzipFile(fileobj=io.BytesIO(response.content)) as gzipped_file:
+            return gzipped_file.read().decode("utf-8")
+
+
+class GEMETSubjectsTransformer(BaseHierarchalSubjectsTransformer):
+    """Transformer class to convert GEMET RDF data to a dictionary format."""
+
+    def _get_parent_notation(self, broader, rdf_graph):
+        """Extract parent notation from GEMET URI."""
+        return "/".join(broader.split("/")[-2:])
+
+    def _get_groups_and_themes(self, subject, rdf_graph):
+        """Extract groups and themes for a subject."""
+        groups = []
+        themes = []
+
+        for relation in rdf_graph.subjects(
+            predicate=self.SKOS_CORE.member, object=subject
+        ):
+            relation_uri = str(relation)
+            relation_label = None
+
+            # If the relation is a group, check for skos:prefLabel
+            if "group" in relation_uri:
+                labels = rdf_graph.objects(
+                    subject=relation, predicate=self.SKOS_CORE.prefLabel
+                )
+                relation_label = next(
+                    (str(label) for label in labels if label.language == "en"), None
```

Comment:
I remember we had an issue with other vocabularies not having the `en` title. Does it apply for this one too?

Reply:
this vocab is very extensive with translations provided, so english translation is always present for each concept, theme and group